sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1442 @@
+            .map(this::convertToTextSentimentResult);
+        Stream<AnalyzeSentimentResult> errorDocumentList = sentimentResponse.getErrors().stream()
+            .map(this::convertToErrorTextSentimentResult);
+
+        return Stream.concat(validDocumentList, errorDocumentList).collect(Collectors.toList());
+    }
+
+    private AnalyzeSentimentResult convertToTextSentimentResult(final DocumentSentiment documentSentiment) {
+        // Document text sentiment
+        final TextSentimentClass documentSentimentClass = convertToTextSentimentClass(documentSentiment.getSentiment());
+        if (documentSentimentClass == null) {
+            return null;
+        }
+
+        final double[] sentimentScores = getTextSentimentScore(documentSentiment.getDocumentScores(),
+            documentSentimentClass);
+
+        // Sentence text sentiment
+        final List<TextSentiment> sentenceSentimentTexts =
+            convertToSentenceSentiments(documentSentiment.getSentences());
+
+        return new AnalyzeSentimentResult(documentSentiment.getId(),
+            documentSentiment.getStatistics() == null ? null
+                : convertToTextDocumentStatistics(documentSentiment.getStatistics()), null,
+            new TextSentiment(documentSentimentClass, sentimentScores[0], sentimentScores[1], sentimentScores[2],
+                sentenceSentimentTexts.stream().mapToInt(TextSentiment::getLength).sum(), 0),
+            sentenceSentimentTexts);
+    }
+
+    private List<TextSentiment> convertToSentenceSentiments(final List<SentenceSentiment> sentenceSentiments) {
+        final List<TextSentiment> sentenceSentimentCollection = new ArrayList<>();
+        sentenceSentiments.forEach(sentenceSentiment -> {
+            final TextSentimentClass sentimentClass = convertToTextSentimentClass(sentenceSentiment.getSentiment());
+            final double[] sentimentScores =
+                getTextSentimentScore(sentenceSentiment.getSentenceScores(), sentimentClass);
+
+            sentenceSentimentCollection.add(new TextSentiment(sentimentClass, sentimentScores[0], sentimentScores[1],
+                sentimentScores[2], sentenceSentiment.getLength(), sentenceSentiment.getOffset()));
+        });
+        return sentenceSentimentCollection;
+    }
+
+    private double[] getTextSentimentScore(SentimentConfidenceScorePerLabel sentimentScore,
+        TextSentimentClass textSentimentClass) {
+        double[] sentimentScores = new double[3];
+        switch (textSentimentClass) {
+            case NEGATIVE:
+                sentimentScores[0] = sentimentScore.getNegative();
+                sentimentScores[1] = 0.0;
+                sentimentScores[2] = 0.0;
+                break;
+            case NEUTRAL:
+                sentimentScores[0] = 0.0;
+                sentimentScores[1] = sentimentScore.getNeutral();
+                sentimentScores[2] = 0.0;
+                break;
+            case POSITIVE:
+                sentimentScores[0] = 0.0;
```

Comment:
Does using sentimentScores.getNegative() for this case return some invalid number? Wondering why we need this switch case here.

Reply:
If the sentiment class is negative, then only negative score value has value other than 0.0. But positive score and neutral score will be 0.0. If the sentiment class is mixed, all three score will have value. This class will be used DocuementSentiment and Sentence sentiment.

Reply:
If that's the case, why don't we just use this snippet instead of the switch case since you know its going to be 0?
```java
sentimentScores[0] = result.getNegative();
sentimentScores[1] = result,getNeutral();
sentimentScores[2] = result.getPositive();
```

Reply:
lol what I had done.